### PR TITLE
Update `react-native-camera-kit` to latest beta to enable programmitcal zooming

### LINKED ISSUE #17358: Zoom controls don't work in chat camera
# Bug Report

## Problem 

The `0,5x`, `1x`, `2x`, `3x` zoom buttons don't work in camera in the chat view. Pinch works fine. 

https://github.com/status-im/status-mobile/assets/67952253/5d7904e8-18eb-456d-a847-ab6bfc4da81c

### Reproduction

1. Open chat
2. Open camera
3. Try to zoom in or out using `2x`, `3x`etc  zoom controls 

### Additional Information

- Status version: nightly 20/09
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #17358

### Summary

Pinch to zoom was already working in the previous version, But we needed to make the buttons zoom work.
And to add that functionality I had to bump react-native-camera-kit.

### Before and after screenshots comparison

| iOS iPhone 13 Pro | Android Pixel 7A
| --- | --- |
| https://github.com/status-im/status-mobile/assets/33176106/cb6ba2a7-60c9-4cfd-be56-77c758be9c23 | https://github.com/status-im/status-mobile/assets/33176106/47a0186e-6273-40b6-8947-67865bc90397 |

status: ready

### CODE REVIEW

File: android/gradle.properties [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 # Version requirements used throughout the Gradle scripts
 kotlinVersion=1.6.20
-minSdkVersion=23
+minSdkVersion=24
```
Comment (ibrkhalil) [inline android/gradle.properties:25]: The minimum SDK of the rn camera kit lib is 24, So I had to update this to match it.

**Comment by ibrkhalil** [comment]
Android is working on this PR, iOS isn't yet.
Continuing investigation tomorrow.

File: ios/Podfile [hunk 2]
```diff
@@ -75,6 +75,12 @@ abstract_target 'Status' do
         # TODO : remove after upgrading to react-native 0.72.x
         config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= ['$(inherited)', '_LIBCPP_ENABLE_CXX17_REMOVED_UNARY_BINARY_FUNCTION']
       end
+      case target.name
+      when 'ReactNativeCameraKit'
+        target.build_configurations.each do |config|
+          config.build_settings.delete 'IPHONEOS_DEPLOYMENT_TARGET'
+        end
+      end
```
Comment (ibrkhalil) [inline ios/Podfile:84]: Needed to remove predefinition of iOS minimum deployment target for RNCameraKit

Comment (siddarthkay) [reply ios/Podfile:84]: Hi @ibrkhalil : Interesting approach. 
Ideally when modifying Podfile of a lib we prefer using a `patchPhase` in nix 
example : https://github.com/status-im/status-mobile/blob/develop/nix/deps/nodejs-patched/default.nix#L92-L99

File: yarn.lock [hunk 3]
```diff
@@ -7018,11 +7018,6 @@ lodash@^3.10.0:
   resolved "https://registry.yarnpkg.com/lodash/-/lodash-3.10.1.tgz#5bf45e8e49ba4189e17d482789dfd15bd140b7b6"
   integrity sha1-W/Rejkm6QYnhfUgnid/RW9FAt7Y=
 
-lodash@^4.14.2:
```
Comment (siddarthkay) [inline yarn.lock:7021]: I'm curious why this happened.
I will verify this locally

File: ios/Podfile (hunk 2 above)
Comment (ibrkhalil) [reply ios/Podfile:84]: Do you think that matching a string in a file and then doing a replacement is better or accessing an attribute in a nested structure and doing a replacement?

I prefer the latter.

But if that's the team's preference happy to do it :)

Comment (siddarthkay) [reply ios/Podfile:84]: There are more ways to patch packages, another one of them is nodejs way of using a patch-package method which is also famous in many libraries.

However in our case, its better to have all the patches live in one place. 
These patches execute in a predictable sequence and hence we follow this approach.

One more example of how we patched `react-native-camera-kit` in upgrade PR : 
https://github.com/status-im/status-mobile/pull/17241/files#diff-b3d31b1df4cceb663a7dba96bcc3ea45686afe2208d7945b6e7e269067d753fbR80-R88

Comment (ibrkhalil) [reply ios/Podfile:84]: Okay, Since this is our way then it makes since to be consistent in how we patch.
Thank you so much, Sid!

Comment (ibrkhalil) [reply ios/Podfile:84]: So I should loop through the inner pods of ReactNativeCameraKit and remove their minimum target uses this way, Okay.

File: yarn.lock (hunk 3 above)
Comment (ibrkhalil) [reply yarn.lock:7021]: lodash is a direct dependency to ReactNativeCameraKit I think
Check in `node_modules/react-native-camera-kit/node_modules/lodash`

Comment (siddarthkay) [reply yarn.lock:7021]: yes this part looks good

File: ios/Podfile (hunk 2 above)
Comment (siddarthkay) [reply ios/Podfile:84]: not in pods,
here : `node_modules/react-native-camera-kit/ios/ReactNativeCameraKit.xcodeproj/project.pbxproj` 
search for "IPHONEOS_DEPLOYMENT_TARGET"

File: nix/deps/nodejs-patched/default.nix [hunk 4]
```diff
@@ -97,6 +98,12 @@ stdenv.mkDerivation {
           '[RCTConvert UIColor:options.cancelButtonTintColor() ? @(*options.cancelButtonTintColor()) : nil];' \
           '[RCTConvert UIColor:options.tintColor() ? @(*options.tintColor()) : nil];'
   '';
+  patchReactNativeCameraKitPhase = ''
```
Comment (jakubgs) [inline nix/deps/nodejs-patched/default.nix]: I see no comment explaining why this is necessary.

**Review by jakubgs — APPROVED** [review]
Ugh, more patching.

**Comment by qoqobolo** [comment]
> Android is working on this PR, iOS isn't yet.
Continuing investigation tomorrow.

Hi @ibrkhalil, thanks for the fix!
Please, ping `@status-im/mobile-qa` once it's ready to be tested on both platforms, thank you!

File: ios/Podfile (hunk 2 above)
Comment (siddarthkay) [reply ios/Podfile:84]: its fine to do this in podfile as well! I have no strong opinion :)

**Review by siddarthkay — APPROVED** [review]

**Comment by ibrkhalil** [comment]
> > Android is working on this PR, iOS isn't yet.
> > Continuing investigation tomorrow.
> 
> Hi @ibrkhalil, thanks for the fix! Please, ping `@status-im/mobile-qa` once it's ready to be tested on both platforms, thank you!

@status-im/mobile-qa Ready for review :)

File: ios/Podfile (hunk 2 above)
Comment (jakubgs) [reply ios/Podfile:84]: I think Podfile change is less fragile than substituting strings.

**Comment by qoqobolo** [comment]
Thanks for the fix @ibrkhalil, PR can be merged.

**Pull request merged**